Source/Blazorise.AntDesign/Badge.razor

```diff
@@ -5,9 +5,9 @@
         <span>@ChildContent</span>
         @if ( Closable )
         {
-            <span role="img" aria-label="close" tabindex="-1" class="@CloseClassBuilder.Class" @onclick="@OnCloseClickedHandler" @onclick:preventDefault>
+            <a href="#" role="img" aria-label="close" class="@CloseClassBuilder.Class" @onclick="@OnCloseClickedHandler" @onclick:preventDefault>
```

Comment:
Is there a reason why you use a `<a>` element?

Reply:
because it accepts focus so we can tab to it

Reply:
Yes, but it is not according to Bootstrap specifications. https://getbootstrap.com/docs/5.2/components/badge/

Also, it might have some unintentional side effects; for example, if users have custom CSS for the regular `a` element, then they would have to adopt it for badges.

Reply:
BTW I have realized that you placed `a` inside of another `a`, which is not valid according to HTML specs https://stackoverflow.com/questions/7863554/is-it-ok-to-have-an-a-inside-another-a

Reply:
> On another note, there's actually a bigger issue with autocomplete - the listbox does not scroll

Yeah, we already had that reported.

Reply:
PS. `span` can also accept tab focus https://jsfiddle.net/westonruter/UAWRc/

Reply:
> which is not valid according to HTML specs

Ok, yes, I wasn't happy about nested <a> but it works so I left it like that for this prototype. But it sounds like either autocomplete needs to handle focus manually or Badge is not the appropriate component for this and it needs to be customized for autocomplete. Using something like tabindex on a span also feels hacky, and probably doesn't support accessibility as well as <a> but I am not sure about that part. At any rate, thought I would share this code because I recently implemented a similar component for my hobby project and it worked well there.

Reply:
> > On another note, there's actually a bigger issue with autocomplete - the listbox does not scroll
> 
> Yeah, we already had that reported.

Fixed on some auto complete PR don't remember which. Might be on the one that's open.

Reply:
It's along one of these commits 1de6a5132232ae7acd0834ae085e1c6e64f719f9

Reply:
Please ignore my last commit then :)

Reply:
> It's along one of these commits [1de6a51](https://github.com/Megabit/Blazorise/commit/1de6a5132232ae7acd0834ae085e1c6e64f719f9)

Actually, I just checked out this branch, scrolling still doesn't work 

https://user-images.githubusercontent.com/22550674/180106813-69859bed-9ae8-4982-bcaf-5b14d5bbb018.mp4

Reply:
We changed it to grey, to diferetiante it between actual selected items and just hover selection. I'll take a look if it's nt working for some reason.

Reply:
> Actually, I just checked out this branch, scrolling still doesn't work

Just tested it again, looks like there's an issue with scrolling only in Chrome/Edge but Firefox works fine. In Chrome/Edge it scrolls down but not up for some reason unless you hit the top element

https://user-images.githubusercontent.com/22550674/180428642-26cb7d23-1095-4e99-8035-e929a581f002.mp4

Reply:
Thanks for testing

Reply:
Can you re-explain how you provoke that bug? The one where it does not scroll back up?

Reply:
Edge:
![chrome-capture-2022-6-23](https://user-images.githubusercontent.com/22283161/180600673-8a158818-89eb-4df4-a5ab-711f2fb5833b.gif)

Reply:
> We changed it to grey as it's the same as regular hover

Got it. I feel since it has checkboxes, it should not highlight selected items in blue, and have a regular blue active item (rather than selected item), imho, not sure if that's standard. I checked the `<select>` behavior it highlights the selected items and then you can move with ctrl-arrow.. But select does not use checkboxes...

Reply:
> Got it. I feel since it has checkboxes, it should not highlight selected items in blue, and have a regular blue active item (rather than selected item), imho, not sure if that's standard. 


Sorry I don't follow? You're suggesting that it should not have the blue selection look? 
(Also the checkboxes are a new option that's active, it also works without checkboxes) 

The grey is in line with the regular hover on Dropdowns, that's why we changed it (Just that they don't have arrow up/down support yet)
![image](https://user-images.githubusercontent.com/22283161/180602951-0c6bd1c1-c17c-43f8-87e1-01b81ae33744.png)

And when you actually select something, usually there's an active class that's usually the primary color.

Reply:
> Can you re-explain how you provoke that bug?

Just synced to dev-1.1.0-AutoCompleteMultipleCheckbox and run the demo, no other changes. Tried Bluma and Bootstrap, both repro (does not scroll up) on Edge

<img width="151" alt="image" src="https://user-images.githubusercontent.com/22550674/180602998-0c220b6b-c67f-4233-9250-48202982c10f.png">

Reply:
Damn, what the heck... I might need your help to figure out why we have different behaviours, let's take this discussion over to the other PR, since it has gone out of this PR's scope.

Reply:
> You're suggesting that it should not have the blue selection look?

Yes, but that's just my idea, if there is some standard that we can use it's better to use the standard. I thought we should use blue highlight for active items (clicked on or selected with keyboard), grey for mouse hover, and checkbox to indicate selected items (not highlight them except currently selected one), again, just my personal preference, not per spec :)

Reply:
I still don't fully get haha currently we have
Hover : Grey
Up/down arrow / Highlighted Item: Grey
Selected items : Blue
Click / confirm : I think its blue.

Can you now list your suggestion color?

Pretty sure we need a different color for both the currently highlighted item and the currently selected items, where the highlighted item color has priority over the selected item so you can tell "where you are"

Reply:
Hover: Grey
Up/down arrow / Highlighted Item: Blue
Selected items : No highlight, only Checkbox, (maybe Grey is actually better than none)
Click/Confim: Blue if i understand what you mean

Agree about themes, instead of blue/grey we have Active/Selected/Hover (active is the one you click on or select with up/down, selected is the one with checked checkbox, and hover is mouse hover)

Reply:
> Selected items : No highlight, only Checkbox, (or maybe Grey is actually better than none)

Checkbox is a new mode, there's a mode without checkboxes that still shows selected items, you always need the color to have a better understanding of what's selected.


So I was investigating **BS5** to see if I find similar components to see what they do, since we do base off providers and should try to follow their conventions (which in turn(hopefully) already do follow html standard conventions and such...).

They do have arrow up/down support and they do use the same hover styling. And for Active/Click blue. 
This when looking at dropdowns.
https://getbootstrap.com/docs/5.2/components/dropdowns/
![chrome-capture-2022-6-23](https://user-images.githubusercontent.com/22283161/180610831-665bdf29-b73e-42e6-9498-13a3fc6bf500.gif)

Now if we take a look at something like tabs, where the UP/Down actually is pretty much just changing the current selection then we see blue.
![chrome-capture-2022-6-23 (1)](https://user-images.githubusercontent.com/22283161/180610874-f310ae5f-86e6-4901-bd81-dec7b7b7473b.gif)

I guess the big difference is whether up or down is actually selecting something right away or just trasversing/highlighting a possible list of options just like hover does.

So I'd still have to say the current color scheme is okay,
Any toughts @stsrki ?

Reply:
> you always need the color to have a better understanding of what's selected

For example for low vision/color blind people it's actually the checkbox that serves as the indicator. (Sorry I just had some small experience with Accessibility some time ago and now I think more about it, and i think accessibility support can be a differentiating feature. Some applications, like email clients, must support accessibility by law to be used in government organizartions for example, so if Blazorise can put a bullet point in the list of features like "Accessible components" it can make it a choice over other libraries. I think Accessibility, even it's not used by most of the people, puts the product in  different league). 

For the autocomplete, I think the question is how to behave in multi select mode, I think single select mode is clear. Here's a table of keyboard behavior between current implementation, `<select>` and my proposal
<img width="829" alt="image" src="https://user-images.githubusercontent.com/22550674/180615422-96e327cb-448a-434d-92bc-35c09312bd45.png">

Ideally multi select would also support Shift-Click and Ctrl-Click to select range and individual item respectively, but I think that's extra.

Reply:
> So I'd still have to say the current color scheme is okay

Yes, I think the colors themselves are fine, I just thought it should be the opposite, active element should be Blue (like in single select) and all selected elements should be grey + checkmark.. Even though I think it's more intuitive, you may be right because `<Select>` also highlights the selected ones and not the active one

https://user-images.githubusercontent.com/22550674/180616670-805ee72e-2dcf-46ac-a9c3-20997799faec.mp4

Reply:
Just my 2 cents. Selected element should be with blue color(or any other that is specific for the app theme), and navigation and up/down should have a gray color.